core/IO.carp

```diff
@@ -72,9 +74,47 @@
                                (IO.fclose f)
                                (Result.Success (String.from-chars &r)))))))
 
+  (doc      read-file "returns the contents of a file passed as argument as a string.")
+  (register read-file (Fn [&String] String))
+  (doc  safe-read-file "Reads the content of a file into a (Result String String).\nIt is intended for text files, since the way to determine the length of a String is to use strlen() which probably will be inaccurate for binaries:\n(safe-read-file filename).")
```

Comment:
Typically our convention is to give the safe variants of functions shorter names, reserving `unsafe-*` to denote unsafe versions (see e.g. `Maybe.unsafe-from`).

Maybe call `safe-read-file` just `read-file` and `read-file` `unsafe-read-file`? Another option is to put the unsafe version in an "Unsafe" module so that one has to call it using `IO.Unsafe.read-file`.

Reply:
Ok, breaking change it is. It's done now.
Thank you!